src/libs/Notification/PushNotification/index.native.ts

```diff
@@ -1,57 +1,57 @@
-import Airship, {EventType} from '@ua/react-native-airship';
-import lodashGet from 'lodash/get';
+import Airship, {EventType, PushPayload} from '@ua/react-native-airship';
 import Onyx from 'react-native-onyx';
-import _ from 'underscore';
 import Log from '@libs/Log';
-import * as PushNotification from '@userActions/PushNotification';
+import * as PushNotificationActions from '@userActions/PushNotification';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ForegroundNotifications from './ForegroundNotifications';
-import NotificationType from './NotificationType';
+import NotificationType, {NotificationData} from './NotificationType';
+import PushNotificationType, {ClearNotifications, Deregister, Init, OnReceived, OnSelected, Register} from './types';
+
+type NotificationEventActionCallback = (data: NotificationData) => void;
 
 let isUserOptedInToPushNotifications = false;
 Onyx.connect({
     key: ONYXKEYS.PUSH_NOTIFICATIONS_ENABLED,
-    callback: (val) => (isUserOptedInToPushNotifications = val),
+    callback: (val) => (isUserOptedInToPushNotifications = val ?? false),
 });
 
-const notificationEventActionMap = {};
+const notificationEventActionMap: Partial<Record<EventType, Record<string, NotificationEventActionCallback>>> = {};
 
 /**
  * Handle a push notification event, and trigger and bound actions.
- *
- * @param {String} eventType
- * @param {Object} notification
  */
-function pushNotificationEventCallback(eventType, notification) {
-    const actionMap = notificationEventActionMap[eventType] || {};
-    let payload = lodashGet(notification, 'extras.payload');
+function pushNotificationEventCallback(eventType: EventType, notification: PushPayload) {
+    const actionMap = notificationEventActionMap[eventType] ?? {};
+    let payload = notification.extras.payload;
```

Comment:
```suggestion
    let payload = notification.extras.payload as NotificationData;
```

Reply:
I think is not correct to assert to `NotificationData` at this point, since `payload` can be a `string` and a few lines below we parse it to JSON.

Reply:
+1. The payload is usually a string on Android and an object on iOS